src/com/ms/silverking/cloud/dht/client/impl/OpSender.java

```diff
@@ -216,15 +217,15 @@ private void createMessages(AsyncOperationImpl[] asyncOpImpls, int startIndex, i
         protoMG = asyncOpImpls[startIndex].createProtoMG(estimate);
         
         if (debug) {
-            System.out.printf("asyncOpImpls.length %d startIndex %d endIndex %d\n", asyncOpImpls.length, startIndex, endIndex);
+            Log.fineAsyncf("asyncOpImpls.length %d startIndex %d endIndex %d", asyncOpImpls.length, startIndex, endIndex);
         }
         messageGroups = new ArrayList<>();
         // Walk through all of the compatible operations.
         // For each operation, add to the current protoMG when possible. If not possible,
         // then create a new protoMG and add the old to the messageGroups list.
         for (int i = startIndex; i <= endIndex; i++) {
             if (debug) {
-                System.out.printf("Calling create with %s\tfor%s\n", protoMG.getUUID(), asyncOpImpls[i].objectToString());
+                Log.fineAsyncf("Calling create with %s for %s", protoMG.getUUID(), asyncOpImpls[i].objectToString());
```

Comment:
debug is off by default. Make Log.warningAsyncf so that the debug flag has an effect